src/core/Memory.test.ts

```diff
@@ -1,12 +1,23 @@
-import {getValues, saveValues} from "./Memory";
+import {getValues, saveValues, storageKey} from "./Memory";
+
+beforeEach(() => {
+    localStorage.clear();
+})
 
 test("Test That Value Retrieved will Be Empty If not Set", () => {
     expect(getValues()).toEqual([]);
 })
 
-test("Test that Value Can Be Saved And Retrieved", () => {
+test('Test That Value Will Be Saved In LocalStorage', () => {
     const value = [{date: new Date(), value: 21}, {date: new Date(), value: 48}];
     saveValues(value);
-    expect(getValues()).toEqual(value);
+    expect(localStorage.setItem).toHaveBeenCalledWith(storageKey, JSON.stringify(value));
+    expect(localStorage.__STORE__[storageKey]).toBe(JSON.stringify(value));
+    expect(Object.keys(localStorage.__STORE__).length).toBe(1);
 })
 
+test("Test that Value Can Be Saved And Retrieved", () => {
+    const value = [{date: new Date(), value: 21}, {date: new Date(), value: 48}];
+    saveValues(value);
+    expect(getValues()).toEqual(JSON.parse(JSON.stringify(value)));
```

Comment:
it would be nice to make it parse dates on retrieval, so same data goes in and same data goes out.
I'd recommend to either:
- do it simply just for this case.
- create recursive function that will replace strings that match the date by Date objects
- find some nice small library that can do it for us

Reply:
after a little thought it is actually a little cheat :o 
i'll improve it

Reply:
well, there is not much abstraction in this solution, but it's probably the simplest possible operation. I assume input comes always right